types/validation.d.ts

```diff
@@ -2,32 +2,32 @@ declare module 'mongoose' {
 
   type SchemaValidator<T> = RegExp | [RegExp, string] | Function | [Function, string] | ValidateOpts<T> | ValidateOpts<T>[];
 
-  interface ValidatorProps {
+  interface ValidatorProps<T> {
     path: string;
-    value: any;
+    value: T;
   }
 
-  interface ValidatorMessageFn {
-    (props: ValidatorProps): string;
+  interface ValidatorMessageFn<T> {
+    (props: ValidatorProps<T>): string;
   }
 
-  interface ValidateFn<T> {
-    (value: T, props?: ValidatorProps & Record<string, any>): boolean;
+  interface ValidateFn<T, M = Model<unknown>> {
+    (this: M, value: T, props?: ValidatorProps<T> & Record<string, any>): boolean;
```

Comment:
`this` should be a HydratedDocument, not `Model`. Remember `Model` is a type, so `this` would be a model class, not an instance of a model class. I'd change the generic def from `M = Model<unknown>` to `THydratedDocumentType = HydratedDocument<unknown>`